packages/host/app/components/operator-mode/stack-item.gts

```diff
@@ -412,23 +415,29 @@ export default class OperatorModeStackItem extends Component<Signature> {
     this.containerEl.scrollTop = 0;
   });
 
-  private doCloseAnimation = dropTask(async () => {
-    this.isClosing = true;
-    if (this.itemEl) {
+  private startAnimation = dropTask(
+    async (animationType: 'closing' | 'movingForward') => {
+      this.animationType = animationType;
+      if (!this.itemEl) return;
       await new Promise<void>((resolve) => {
-        scheduleOnce('afterRender', this, this.handleCloseAnimation, resolve);
+        scheduleOnce(
+          'afterRender',
+          this,
+          this.handleAnimationCompletion,
+          resolve,
+        );
       });
-    }
-  });
+    },
+  );
 
-  private handleCloseAnimation(resolve: () => void) {
+  private handleAnimationCompletion(resolve: () => void) {
     if (!this.itemEl) {
       return;
     }
     const animations = this.itemEl.getAnimations() || [];
-    Promise.all(animations.map((animation) => animation.finished)).then(() =>
-      resolve(),
-    );
+    Promise.all(animations.map((animation) => animation.finished))
+      .then(() => resolve())
+      .catch(() => resolve()); // Task was cancelled, which is fine to resolve - eg: the component was destroyed
```

Comment:
This catch is to handle error occurs in test only

`AbortError: The user aborted a request.`

Reply:
Two things on this: 

1) I suspect that a catch here would be related to task cancellation. This promise is purely related to the DOM (`Element#getAnimations()#finished`). Maybe there is something that happens when an animation is canceled?
2) When you do want to catch a task cancellation, it is important to inspect the error in the code and confirm that it is a task cancelation so you don't accidentally catch and ignore real errors.

Reply:
Ah, I found the root issue! It was due to the sequence in `dismissStackedCardsAbove`:

Initially, I implemented the method in this order, which broke the animation:

1. Animate closing items
2. Close items
3. Animate the next top item moving forward

This caused task cancellation because the component was destroyed before Step 3 was complete.

--- 

With the latest changes, no more test errors occur:

1. Animate closing items
2. Animate the next top item moving forward
3. Close items